collagraph/__init__.py

```diff
@@ -583,7 +583,12 @@ def equivalent_code(a, b):
     ]:
         attr_a = getattr(a, attr, None)
         attr_b = getattr(b, attr, None)
-        if attr_a != attr_b:
+        if attr in ["co_freevars", "co_varnames"]:
+            # co_varnames and co_freevars can contain names, which might
+            # be different but should be similar at least in length
+            if len(attr_a) != len(attr_b):
```

Comment:
I don't really understand why using `len()` is sufficient here?

Reply:
The contents of the `co_freevars` and `co_varnames` contain names for variables that are used within the function. A function that is similar in logic but has different variable names is still equivalent, so we can't check for strict equality. If the length is not the same, then we already know that the logic of the functions can't be equal, so we can bail.

Reply:
Ah, so it's just an early exit?

Reply:
Exactly!